src/Proxy/ProxyFactory.php

```diff
@@ -263,7 +265,11 @@ private function getProxyFactory(string $className): Closure
                 $name = $property->name;
 
                 if ($property->isStatic() || (($class->hasField($name) || $class->hasAssociation($name)) && ! isset($identifiers[$name]))) {
-                    continue;
+                    if (version_compare(PHP_VERSION, '8.4', '<')) {
+                        continue;
+                    } elseif ($property->isPrivateSet() === false) {
+                        continue;
+                    }
```

Comment:
This logic looks wrong, it only skips private set fields in PHP 8.4+ - but not others anymore.

Should maybe be:

```php
if ($property->isStatic() || (($class->hasField($name) || $class->hasAssociation($name)) && ! isset($identifiers[$name])) || (PHP_VERSION_ID >= 80400 && $property->isPrivateSet())) {
    continue;
}
```